Mathlib/CategoryTheory/FiberedCategory/Cocartesian.lean

```diff
@@ -0,0 +1,329 @@
+    rintro τ ⟨hτ₁, hτ₂⟩
+    apply map_uniq
+    simp only [Iso.eq_inv_comp, ← assoc, hτ₂]
+
+end IsCocartesian
+
+namespace IsStronglyCocartesian
+
+section
+
+variable {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) [IsStronglyCocartesian p f φ]
+
+/-- The universal property of a strongly cocartesian morphism.
+
+This lemma is more flexible with respect to non-definitional equalities than the field
+`universal_property'` of `IsStronglyCocartesian`. -/
+lemma universal_property {S' : 𝒮} {b' : 𝒳} (g : S ⟶ S') (f' : R ⟶ S') (hf' : f' = f ≫ g)
+    (φ' : a ⟶ b') [IsHomLift p f' φ'] : ∃! χ : b ⟶ b', IsHomLift p g χ ∧ φ ≫ χ = φ' := by
+  subst_hom_lift p f' φ'; clear a b R S
+  have : p.IsHomLift (f ≫ g) φ' := (hf' ▸ inferInstance)
+  apply IsStronglyCocartesian.universal_property' f
+
+instance isCocartesian_of_isStronglyCocartesian [p.IsStronglyCocartesian f φ] :
+    p.IsCocartesian f φ where
+  universal_property := fun φ' => universal_property p f φ (𝟙 S) f (comp_id f).symm φ'
+
+section
+
+variable {S' : 𝒮} {b' : 𝒳} {g : S ⟶ S'} {f' : R ⟶ S'} (hf' : f' = f ≫ g) (φ' : a ⟶ b')
+  [IsHomLift p f' φ']
+
+/-- Given a diagram
+```
+a --φ--> b        b'
+|        |        |
+v        v        v
+R --f--> S --g--> S'
+```
+such that `φ` is strongly cocartesian, and a morphism `φ' : a ⟶ b'`. Then `map` is the map
+`b ⟶ b'` lying over `g` obtained from the universal property of `φ`. -/
+noncomputable def map : b ⟶ b' :=
+  Classical.choose <| universal_property p f φ _ _ hf' φ'
+
+instance map_isHomLift : IsHomLift p g (map p f φ hf' φ') :=
+  (Classical.choose_spec <| universal_property p f φ _ _ hf' φ').1.1
+
+@[reassoc (attr := simp)]
+lemma fac : φ ≫ (map p f φ hf' φ') = φ' :=
+  (Classical.choose_spec <| universal_property p f φ _ _ hf' φ').1.2
+
+
+/-- Given a diagram
+```
+a --φ--> b        b'
+|        |        |
+v        v        v
+R --f--> S --g--> S'
+```
+such that `φ` is strongly cocartesian, and morphisms `φ' : a ⟶ b'`, `ψ : b ⟶ b'` such that
+`g ≫ ψ = φ'`. Then `ψ` is the map induced by the universal property. -/
+lemma map_uniq (ψ : b ⟶ b') [IsHomLift p g ψ] (hψ : φ ≫ ψ = φ') : ψ = map p f φ hf' φ' :=
+  (Classical.choose_spec <| universal_property p f φ _ _ hf' φ').2 ψ ⟨inferInstance, hψ⟩
+
+end
+
+/-- Given a diagram
+```
+a --φ--> b        b'
+|        |        |
+v        v        v
+R --f--> S --g--> S'
+```
+such that `φ` is strongly cocartesian, and morphisms `ψ ψ' : b ⟶ b'` such that
+`g ≫ ψ = φ' = g ≫ ψ'`. Then we have that `ψ = ψ'`. -/
+protected lemma ext {S' : 𝒮} {b' : 𝒳} (g : S ⟶ S') {ψ ψ' : b ⟶ b'} [IsHomLift p g ψ]
+    [IsHomLift p g ψ'] (h : φ ≫ ψ = φ ≫ ψ') : ψ = ψ' := by
+  rw [map_uniq p f φ (g := g) rfl (φ ≫ ψ) ψ rfl, map_uniq p f φ (g := g) rfl (φ ≫ ψ) ψ' h.symm]
+
+@[simp]
+lemma map_self : map p f φ (comp_id f).symm φ = 𝟙 b := by
+  subst_hom_lift p f φ; symm
+  apply map_uniq
+  simp only [comp_id]
+
+/-- The composition of two `IsCocartesian.map` is also given by `IsCocartesian.map`. -/
```

Comment:
Could you add diagrams to this docstring like in the other file?

Reply:
I added them for strongly cartesian / cocartesian, for cartesian / cocartesian I just added a text explanation as it did not seem much more helpful with a plaintext diagram.